startupscript/butane/idle-shutdown.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+# idle-shutdown.sh shuts down the VM if it has been idle for a certain amount of time. This script is
+# run on the host VM by a systemd timer unit to check for inactivity and shut down the VM. The script
+# will be run n seconds after boot up and run every m minutes. By default, it will start checking system 
+# idleness 48 hours after boot and run every 5 minutes.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o xtrace
+
+function emit() {
+  echo "$(date '+%Y-%m-%d %H:%M:%S') $*"
+}
+readonly -f emit
+
+# shellcheck source=/dev/null
+source /home/core/metadata-utils.sh
+
+# Get the idle timeout in seconds. By default, the VM timeout after 2 days of continued idleness.
+IDLE_TIMEOUT_SECONDS="${1:-172800}"
+
+# Get the last time the VM was active.
+LAST_ACTIVE="$(get_guest_attribute "last-active/cpu" || echo "0")"
+LAST_ACTIVE_PROXY="$(get_guest_attribute "last-active/proxy" || echo "0")"
+
+# get the latest time between the two
+if [[ "${LAST_ACTIVE}" -lt "${LAST_ACTIVE_PROXY}" ]]; then
+    LAST_ACTIVE="${LAST_ACTIVE_PROXY}"
+fi
+readonly LAST_ACTIVE
+emit "Last active time: ${LAST_ACTIVE}"
+
+NOW="$(date +'%s')"
+readonly NOW
+
+# Check if the VM has been idle for longer than the timeout.
+if [[ $((NOW - LAST_ACTIVE)) -gt IDLE_TIMEOUT_SECONDS ]]; then
+    emit "Shutting down the VM. Last active time: ${LAST_ACTIVE}. Idle timeout threshold is: ${IDLE_TIMEOUT_SECONDS}"
```

Comment:
I think it is still worth including NOW in the message, since it's a numeric value and helps the reader check the calculation.